### ISSUE TITLE
[pistache] Fatal Crash when the rest client does not consume the whole body

### ISSUE THREAD

**Issue by ronanjs** [Spirent/openperf#358]
Happens if the client closes the connection before the whole body has been sent. Can be reproduce for instance with `curl`:

```
curl -I "http://10.109.4.112:9000/packet/analyzer-results" -X GET     10:33:39
HTTP/1.1 200 OK
Content-Type: application/json
Connection: Close
Content-Length: 87595
```

Call stack

```
Thread 65 "op_api" received signal SIGPIPE, Broken pipe.
[Switching to Thread 0x7ffea2ffd700 (LWP 21566)]
0x00007ffff700c9be in __libc_send (fd=348, buf=0x7ffe9806674c, len=8156, flags=0)
    at ../sysdeps/unix/sysv/linux/send.c:28
28	../sysdeps/unix/sysv/linux/send.c: No such file or directory.
(gdb) bt
#0  0x00007ffff700c9be in __libc_send (fd=348, buf=0x7ffe9806674c, len=8156, flags=0)
    at ../sysdeps/unix/sysv/linux/send.c:28
#1  0x0000000000e12b1a in asyncWriteImpl () at /root/project/deps/pistache/src/common/transport.cc:254
#2  0x0000000000e11996 in onReady () at /root/project/deps/pistache/src/common/transport.cc:111
#3  0x0000000000e17af2 in handleFds () at /root/project/deps/pistache/src/common/reactor.cc:186
#4  runOnce () at /root/project/deps/pistache/src/common/reactor.cc:148
#5  0x0000000000e1819c in run () at /root/project/deps/pistache/src/common/reactor.cc:159
#6  0x0000000000e18de4 in operator() () at /root/project/deps/pistache/src/common/reactor.cc:472
#7  __invoke_impl<void, (lambda at /root/project/deps/pistache/src/common/reactor.cc:468:34)> ()
    at /usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/invoke.h:60
#8  __invoke<(lambda at /root/project/deps/pistache/src/common/reactor.cc:468:34)> ()
    at /usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/invoke.h:95
#9  _M_invoke<0> () at /usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/thread:244
#10 operator() () at /usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/thread:253
#11 _M_run () at /usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/thread:196
#12 0x0000000000d6a79f in execute_native_thread_routine ()
#13 0x00007ffff70026db in start_thread (arg=0x7ffea2ffd700) at pthread_create.c:463
#14 0x00007ffff6d2ba3f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

**Comment by ronanjs** [comment]
Can still be reproduced with the latest version using 

```
curl -X GET -I "http://10.61.35.222:9000/packet/analyzer-results/"                                                                                                                               HTTP/1.1 200 OK
Content-Type: application/json
Connection: Close
Transfer-Encoding: chunked
```

Stack:

```
Thread 25 "op_api" received signal SIGPIPE, Broken pipe.
[Switching to Thread 0x7fff7affd700 (LWP 24601)]
__libc_send (flags=0, len=50308, buf=0x7fff581b0c85, fd=2187) at ../sysdeps/unix/sysv/linux/send.c:28
28	../sysdeps/unix/sysv/linux/send.c: No such file or directory.
(gdb) bt
#0  __libc_send (flags=0, len=50308, buf=0x7fff581b0c85, fd=2187) at ../sysdeps/unix/sysv/linux/send.c:28
#1  __libc_send (fd=2187, buf=0x7fff581b0c85, len=50308, flags=0) at ../sysdeps/unix/sysv/linux/send.c:23
#2  0x0000000000eaf802 in asyncWriteImpl () at /root/project/deps/pistache/src/common/transport.cc:254
#3  0x0000000000eae67e in onReady () at /root/project/deps/pistache/src/common/transport.cc:111
#4  0x0000000000eb47b6 in handleFds () at /root/project/deps/pistache/src/common/reactor.cc:186
#5  runOnce () at /root/project/deps/pistache/src/common/reactor.cc:148
#6  0x0000000000eb4e60 in run () at /root/project/deps/pistache/src/common/reactor.cc:159
#7  0x0000000000eb5aa8 in operator() () at /root/project/deps/pistache/src/common/reactor.cc:472
#8  __invoke_impl<void, (lambda at /root/project/deps/pistache/src/common/reactor.cc:468:34)> () at /usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/invoke.h:60
#9  __invoke<(lambda at /root/project/deps/pistache/src/common/reactor.cc:468:34)> () at /usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/invoke.h:95
#10 _M_invoke<0> () at /usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/thread:244
#11 operator() () at /usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/thread:253
#12 _M_run () at /usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/thread:196
#13 0x0000000000e0476f in execute_native_thread_routine ()
#14 0x00007ffff7df8fa3 in start_thread (arg=<optimized out>) at pthread_create.c:486
#15 0x00007ffff7d274cf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

**Comment by DerangedMonkeyNinja** [comment]
Should be resolved by d474f1e5b517be817ea295f0ebe3c714bcfb1c3a
